### PR TITLE
chore(ci): improve visibility of failed backport bot merges

### PR DESCRIPTION
**Pull request by pluveto**
#### Background 

The failure comment like this [one](https://github.com/Kong/kong-ee/pull/4408#issuecomment-1404006790) is easy to miss.

These failures should be very noisy to prevent cases where a fix is not carried across the required release branches.

#### This PR

Sends a Slack message to `gateway-notifications` channel when a 'team-gateway-bot' makes a comment containing 'To backport manually, run these commands in your terminal' on an PR comment issue.

image

#### Why new action

It is impossible to hook the backport fail event because the backport action doesn't provide that output.

https://github.com/tibdex/backport/blob/main/src/backport.ts

So we created this new workflow.

### Checklist

- [ ] The Pull Request has tests
- [ ] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* [Implement ...]

### Issue reference

Fix KAG-849

### CODE REVIEW

**Comment by pluveto** [comment]
`SLACK_WEBHOOK_URL` Need to be configured

**Comment by pluveto** [comment]
Now support mention users. @dndx need set `vars.SLACK_MAPPING`

**Comment by RobSerafini** [comment]
@kikito @tyler-ball - please recommend a reviewer for the PR.

File: .github/workflows/backport-fail-bot.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            const slack_mapping = JSON.parse(process.env.SLACK_MAPPING);
+            const pr_url = "${{ github.event.issue.pull_request.html_url}}";
+            const pr_author_github_id = "${{ github.event.issue.user.login }}"
+            const pr_author_slack_id = slack_mapping[pr_author_github_id];
+            const author = (pr_author_slack_id ? `<@${pr_author_slack_id}>` : pr_author_github_id);
+            const payload = {
+              text: `Backport failed in PR: ${pr_url}. ${author} check it.`,
+              channel: process.env.SLACK_CHANNEL,
+            };
+            console.log(`::set-output name=payload::${JSON.stringify(payload)}`);
```
Comment (ADD-SP) [inline .github/workflows/backport-fail-bot.yml:26]: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

The `::set-output` has been deprecated, could you please use the newer way?

**Review by ADD-SP — CHANGES_REQUESTED** [review]

**Review by ADD-SP — APPROVED** [review]

**Comment by hbagdi** [comment]
Why not create a Jira and assign the PR author?
Slack is yet another place where these failures will be ignored, no?

**Comment by pluveto** [comment]
> Why not create a Jira and assign the PR author? Slack is yet another place where these failures will be ignored, no?

I think we should do experiments before drawing conclusions

**Comment by kikito** [comment]
I have assigned @samugi on this one as a reviewer

**Comment by samugi** [comment]
A comment in the thread above says: `@dndx need set vars.SLACK_MAPPING`, is this action still pending / is that the reason for the "do-not-merge" label @dndx @pluveto?

**Comment by pluveto** [comment]
> A comment in the thread above says: `@dndx need set vars.SLACK_MAPPING`, is this action still pending / is that the reason for the "do-not-merge" label @dndx @pluveto?

Yes

**Comment by chronolaw** [comment]
This PR should not be included in milestone 3.3.

**Comment by dndx** [comment]
This PR is out of date and some variable names has since changed. I need to work on fixing it before merging.

**Comment by ADD-SP** [comment]
Close and move to https://github.com/Kong/kong/pull/10894

**Pull request closed without merging**